### ISSUE TITLE
Configure TrackEvent in UI

### ISSUE THREAD

**Issue by rickvanprim** [google/perfetto#155]
It would be helpful to be able to configure one or more TrackEvent data sources in the Perfetto UI.  Ideally the UI would have the option to query the current set of Producers to get a list of available processes, categories, and tags, and an easy way to filter accordingly.

Currently to use TrackEvent data sources, the trace must be executed outside of the UI in order to supply a handwritten TraceConfig.

**Comment by betasheet** [comment]
Agree, this should definitely be on our radar :) Technically, the querying described by @rickvanprim should be possible after @octaviantu's improvements. We can already query the [track event data source descriptor](https://cs.android.com/android/platform/superproject/+/master:external/perfetto/protos/perfetto/common/track_event_descriptor.proto;l=28?q=TrackEventDescriptor&ss=android%2Fplatform%2Fsuperproject) with available categories from the service. We'd probably want to extend it such that it also specifies available tags.

**Comment by santisan** [comment]
Would be great having the ability to configure track events from the UI. Currently we are forced to write the config separately and invoke from the terminal.

**Comment by LalitMaganti** [comment]
@primiano maybe you can fix this as part of your look again into recording?

**Comment by LalitMaganti** [comment]
Folding this into https://github.com/google/perfetto/issues/2155 which is much nicer written version of this request.
